### PR TITLE
oss backend support profile

### PR DESCRIPTION
**Pull request by YY-tomorrow**

### CODE REVIEW

File: backend/remote-state/oss/backend.go [hunk 1]
```diff
@@ -134,6 +137,16 @@ func New() backend.Backend {
 			},
 
 			"assume_role": assumeRoleSchema(),
+			"shared_credentials_file": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The path to the shared credentials file. If not set this defaults to ~/.aliyun/config.json",
```
Comment (xiaozhu36) [inline backend/remote-state/oss/backend.go]: This is the path to the shared credentials file. If this is not set, ~/.aliyun/config.json will be used.

File: backend/remote-state/oss/backend.go [hunk 2]
```diff
@@ -134,6 +137,16 @@ func New() backend.Backend {
...
+			},
+			"profile": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The profile for API operations. If not set, the default profile created with `aliyun configure` will be used.",
```
Comment (xiaozhu36) [inline backend/remote-state/oss/backend.go]: This is the Alibaba Cloud profile name as set in the shared credentials file. If not set, the default profile created with `aliyun configure` will be used.

File: backend/remote-state/oss/backend.go [hunk 3]
```diff
@@ -159,7 +172,7 @@ func assumeRoleSchema() *schema.Schema {
 					Type:        schema.TypeString,
 					Optional:    true,
 					Description: "The session name to use when assuming the role.",
-					DefaultFunc: schema.EnvDefaultFunc("ALICLOUD_ASSUME_ROLE_SESSION_NAME", "terraform"),
```
Comment (xiaozhu36) [inline backend/remote-state/oss/backend.go:162]: why removed?

File: website/docs/backends/types/oss.html.md [hunk 4]
```diff
@@ -90,6 +90,8 @@ The following configuration options or environment variables are supported:
  * `acl` - (Optional) [Object
    ACL](https://www.alibabacloud.com/help/doc-detail/52284.htm)
    to be applied to the state file.
+ * `shared_credentials_file` - (Optional) This is the path to the shared credentials file. If this is not set and a profile is specified, ~/.aliyun/config.json will be used.
+ * `profile` - (Optional) This is the Alicloud profile name as set in the shared credentials file.   
  * `assume_role` - (Optional) If provided with a role ARN, will attempt to assume this role using the supplied credentials.
```
Comment (xiaozhu36) [inline website/docs/backends/types/oss.html.md]: still applies

**Review by xiaozhu36 — CHANGES_REQUESTED** [review]

File: backend/remote-state/oss/backend.go (hunk 3 above)
Comment (YY-tomorrow) [reply backend/remote-state/oss/backend.go:162]: 默认取terraform，后面的从profile取值，无法判断是默认值还是用户填入的值

File: backend/remote-state/oss/backend.go [hunk 5]
```diff
@@ -151,20 +164,20 @@ func assumeRoleSchema() *schema.Schema {
 			Schema: map[string]*schema.Schema{
 				"role_arn": {
 					Type:        schema.TypeString,
-					Required:    true,
-					Description: "The ARN of a RAM role to assume prior to making API calls.",
+					Optional:    true,
+					Description: "The role to be assumed",
```
Comment (xiaozhu36) [inline backend/remote-state/oss/backend.go]: 这个为啥改成Optional？
Description不用改

File: backend/remote-state/oss/backend.go [hunk 6]
```diff
@@ -151,20 +164,20 @@ func assumeRoleSchema() *schema.Schema {
...
 					Optional:    true,
 					Description: "The session name to use when assuming the role.",
-					DefaultFunc: schema.EnvDefaultFunc("ALICLOUD_ASSUME_ROLE_SESSION_NAME", "terraform"),
+					DefaultFunc: schema.EnvDefaultFunc("ALICLOUD_ASSUME_ROLE_SESSION_NAME", ""),
 				},
 				"policy": {
 					Type:        schema.TypeString,
 					Optional:    true,
-					Description: "The permissions applied when assuming a role. You cannot use this policy to grant permissions which exceed those of the role that is being assumed.",
+					Description: "The permissions applied when assuming a role.",
```
Comment (xiaozhu36) [inline backend/remote-state/oss/backend.go]: 不用改

File: website/docs/backends/types/oss.html.md [hunk 7]
```diff
@@ -90,18 +90,19 @@ The following configuration options or environment variables are supported:
  * `acl` - (Optional) [Object
    ACL](https://www.alibabacloud.com/help/doc-detail/52284.htm)
    to be applied to the state file.
- * `assume_role` - (Optional) If provided with a role ARN, will attempt to assume this role using the supplied credentials.
-
```
Comment (xiaozhu36) [inline website/docs/backends/types/oss.html.md:94]: 不用改

File: website/docs/backends/types/oss.html.md [hunk 8]
```diff
@@ -90,18 +90,19 @@ The following configuration options or environment variables are supported:
...
    to be applied to the state file.
- * `assume_role` - (Optional) If provided with a role ARN, will attempt to assume this role using the supplied credentials.
- 
+ * `shared_credentials_file` - (Optional, Available in 0.12.7+) Path to a shared credentials file
+ * `profile` - (Optional, Available in 0.12.7+) Alibaba Cloud profile name.  
+ * `assume_role` - (Optional, Available in 0.12.6+) If provided with a role ARN, will attempt to assume this role using the supplied credentials.
+
 The nested `assume_role` block supports the following:
-* `role_arn` - (Required) The ARN of the role to assume. If ARN is set to an empty string, it does not perform role switching. It supports environment variable `ALICLOUD_ASSUME_ROLE_ARN`.
-  Terraform executes configuration on account with provided credentials.
```
Comment (xiaozhu36) [inline website/docs/backends/types/oss.html.md]: 不用改

File: website/docs/backends/types/oss.html.md [hunk 9]
```diff
@@ -90,18 +90,19 @@ The following configuration options or environment variables are supported:
...
 
-* `policy` - (Optional) A more restrictive policy to apply to the temporary credentials. This gives you a way to further restrict the permissions for the resulting temporary
-  security credentials. You cannot use this policy to grant permissions which exceed those of the role that is being assumed.
```
Comment (xiaozhu36) [inline website/docs/backends/types/oss.html.md]: 不用改

File: website/docs/backends/types/oss.html.md [hunk 10]
```diff
@@ -90,18 +90,19 @@ The following configuration options or environment variables are supported:
...
+* `role_arn` - (Optional) The role to be assumed. It supports environment variable `ALICLOUD_ASSUME_ROLE_ARN`. Terraform executes configuration on account with provided credentials.
+
+* `policy` - (Optional) The permissions applied when assuming a role.
 
-* `session_name` - (Optional) The session name to use when assuming the role. If omitted, 'terraform' is passed to the AssumeRole call as session name. It supports environment variable `ALICLOUD_ASSUME_ROLE_SESSION_NAME`.
+* `session_name` - (Optional) The session name to use when assuming the role. It supports environment variable `ALICLOUD_ASSUME_ROLE_SESSION_NAME`.
```
Comment (xiaozhu36) [inline website/docs/backends/types/oss.html.md]: 不用改

File: website/docs/backends/types/oss.html.md [hunk 11]
```diff
@@ -90,18 +90,19 @@ The following configuration options or environment variables are supported:
...
 
-* `session_expiration` - (Optional) The time after which the established session for assuming role expires. Valid value range: [900-3600] seconds. Default to 3600 (in this case Alibaba Cloud use own default value). It supports environment variable `ALICLOUD_ASSUME_ROLE_SESSION_EXPIRATION`.
+* `session_expiration` - (Optional) The time after which the established session for assuming role expires. It supports environment variable `ALICLOUD_ASSUME_ROLE_SESSION_EXPIRATION`.
```
Comment (xiaozhu36) [inline website/docs/backends/types/oss.html.md]: 不用改

File: backend/remote-state/oss/backend.go [hunk 12]
```diff
@@ -209,28 +222,58 @@ func (b *Backend) configure(ctx context.Context) error {
 	b.serverSideEncryption = d.Get("encrypt").(bool)
 	b.acl = d.Get("acl").(string)
 
-	accessKey := d.Get("access_key").(string)
-	secretKey := d.Get("secret_key").(string)
-	securityToken := d.Get("security_token").(string)
-	region := d.Get("region").(string)
+	var getConfigFromProfileByKey = func(str string, key string) string {
```
Comment (xiaozhu36) [inline backend/remote-state/oss/backend.go]: getConfigFromProfileByKey 改为 getBackendConfig

**Review by xiaozhu36 — CHANGES_REQUESTED** [review]

**Review by xiaozhu36 — APPROVED** [review]

**Pull request merged**